vault/auth.go

```diff
@@ -608,7 +769,31 @@ func (c *Core) loadCredentials(ctx context.Context) error {
 }
 
 // persistAuth is used to persist the auth table after modification
-func (c *Core) persistAuth(ctx context.Context, table *MountTable, local *bool) error {
+func (c *Core) persistAuth(ctx context.Context, barrier logical.Storage, table *MountTable, local *bool, mount string) error {
+	// Sometimes we may not want to explicitly pass barrier; fetch it if
+	// necessary.
+	if barrier == nil {
+		barrier = c.barrier
+	}
+
+	// Gracefully handle a transaction-aware backend, if a transaction
+	// wasn't created for us. This is safe as we do not support nested
+	// transactions.
+	needTxnCommit := false
+	if _, ok := barrier.(logical.TransactionalStorage); ok {
+		var err error
+		barrier, err = barrier.(logical.TransactionalStorage).BeginTx(ctx)
```

Comment:
nit: defend against assertion safety

Reply:
That is done two lines above. :-) If barrier changes between now and then, we've got bigger problems :-D

Reply:
can't see. maybe make use of val, ok - as you are already doing casting on a previous step.